pkg/util/template2/translation.go

```diff
@@ -0,0 +1,104 @@
+package template
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/authgear/authgear-server/pkg/util/resource"
+	"github.com/authgear/authgear-server/pkg/util/template"
+)
+
+const translationJSONName = "translation.json"
+
+type translationJSON struct{}
+
+var TranslationJSON = resource.RegisterResource(&translationJSON{})
+
+func (t *translationJSON) ReadResource(fs resource.Fs) ([]resource.LayerFile, error) {
+	return readTemplates(fs, translationJSONName)
+}
+
+func (t *translationJSON) MatchResource(path string) bool {
+	return matchTemplatePath(path, translationJSONName)
+}
+
+func (t *translationJSON) Merge(layers []resource.LayerFile, args map[string]interface{}) (*resource.LayerFile, error) {
+	return mergeTranslations(layers, args)
+}
+
+func (t *translationJSON) Parse(data []byte) (interface{}, error) {
+	var translations map[string]template.Translation
```

Comment:
This is a bit confusing. Before `Merge`, the data of LayerFile is `map[string]string`, after Merge is done and result in a single LayerFile, the data becomes `map[string]template.Translation`.

Maybe we should add a distinct type for MergeFile and rename `Parse(data []byte)` to `ParseMerged(resource.MergedFile)`?

Reply:
Agreed, let me change that.